bep032tools/generator/tests/test_BEP032Generator.py

```diff
@@ -184,7 +184,56 @@ def test_generate_example_structure(self):
     def doCleanups(self):
         initialize_test_directory(clean=True)
 
+class Test_FolderGeneration(unittest.TestCase):
+
+    def setUp(self):
+        self.test_dir = Path(initialize_test_directory(clean=True))
+        os.mkdir(self.test_dir / 'generateTest')
+        csv_filename = generate_simple_csv_file()
+        self.csv_file = csv_filename
+
+    def test_generate_folder(self):
+        """
+        When the file did not exist, the generation did not work properly.
+        Now if the folder does not exist, the function creates the folder before the generation function is launched
```

Comment:
This is method here is a test and will not be run when a user is using the BEP032Generator. In your documentation it sounds like you expect this code here to be executed also in the normal user mode. This should be rather something like `Check that generate_struct does also accepts non-existing target folders and generates missing folders correctly.`